src/connectors/webwallet/helpers/trpc.ts

```diff
@@ -53,6 +53,36 @@ export const setPopupOptions = ({
   popupParams = `width=${width},height=${height},top=${y},left=${x},toolbar=no,menubar=no,scrollbars=no,location=no,status=no,popup=1`
 }
 
+const PopupManager = {
+  currentPopup: null as Window | null,
+
+  createPopup(url: string, name: string, params: string): Window {
+    // Close any existing popup
+    if (this.currentPopup && !this.currentPopup.closed) {
+      this.currentPopup.close()
+    }
+
+    // Create popup immediately
+    const popup = window.open(url, name, params)
+
+    if (!popup) {
+      throw new Error("Popup blocked by browser")
+    }
+
+    this.currentPopup = popup
+
+    // Monitor popup state
+    const checkClosed = setInterval(() => {
+      if (popup.closed) {
```

Comment:
could be an edge case, but theoretically `this.currentPopup !== popup` should not happen in this case
popup is always the new one

thinking if this approach could be more reliable but it would requires some tests (using `beta` branch)